test/c/runner.c

```diff
@@ -0,0 +1,162 @@
+#include <libpq-fe.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+// Include your test files here
+#include "test_op_rewrite.c"
+// ===========================
+
+typedef int (*TestCaseFunction)(PGconn *);
+
+struct TestCase
+{
+    char            *name;
+    TestCaseFunction func;
+};
+
+PGconn *connect_database(
+    const char *db_host, const char *db_port, const char *db_user, const char *db_password, const char *db_name)
+{
+    const int const_db_uri_chars = strlen("host= port= user= dbname= sslmode=disable password=");
+    char *db_uri = malloc(strlen(db_host) + strlen(db_port) + strlen(db_user) + strlen(db_password) + strlen(db_name)
+                          + const_db_uri_chars);
+    sprintf(db_uri,
+            "host=%s port=%s user=%s dbname=%s sslmode=disable password=%s",
+            db_host,
+            db_port,
+            db_user,
+            db_name,
+            db_password);
+
+    PGconn *conn = PQconnectdb(db_uri);
+    free(db_uri);
+
+    if(PQstatus(conn) != CONNECTION_OK) {
+        fprintf(stderr, "Connection to database failed: %s\n", PQerrorMessage(conn));
+        PQfinish(conn);
+        return NULL;
+    }
+    return conn;
+}
+
+int recreate_database(PGconn *root_conn, const char *test_db_name)
+{
+    char *statement = "DROP DATABASE IF EXISTS ";
+    char *full_statement = malloc(strlen(statement) + strlen(test_db_name));
+    sprintf(full_statement, "%s%s", statement, test_db_name);
+    PGresult *res = PQexec(root_conn, full_statement);
+    free(full_statement);
+
+    if(PQresultStatus(res) != PGRES_COMMAND_OK) {
+        fprintf(stderr, "Failed to drop test db: %s\n", PQerrorMessage(root_conn));
+        PQclear(res);
+        return 1;
+    }
+
+    statement = "CREATE DATABASE ";
+    full_statement = malloc(strlen(statement) + strlen(test_db_name));
```

Comment:
Should there be a `+1` in malloc size here for `\n` character in sprintf below?

Reply:
True, forgot about nullbyte.